File: 성민/레벨2/짝지어_제거하기.js
function solution(s) {
  let answer = [];
  for (const i of s) {
    if (answer.length === 0) {
      answer.push(i);
    } else {
      if (answer[answer.length - 1] === i) {
        answer.pop();
      } else {
        answer.push(i);
      }
    }
  }
  return answer.length > 0 ? 0 : 1;
}

console.log(solution("abccaeeaba"));

File: 종우/짝지어제거하기.py
# 1012_1
# https://school.programmers.co.kr/learn/courses/30/lessons/12973

def solution(s):
    stack = []
    
    for char in s:
        if len(stack) == 0:
            stack.append(char)
        elif stack[-1] == char:
            stack.pop()
        else:
            stack.append(char)
    if len(stack) == 0:
        return 1
    else:
        return 0